### PR TITLE
type return/break/continue expressions with monomorphs instead of Dynamic (see #10744)

### PR DESCRIPTION
**Pull request by nadako**
this is what we already do for `throw` since 2008 (https://github.com/HaxeFoundation/haxe/commit/9cee2fcc78e7715d69fcdc8dcee36413c052c9e1)

### CODE REVIEW

**Comment by nadako** [comment]
I'm wondering if we should be a bit more clever and type all terminators as `Void` if they are used in `WithType.NoValue` context. Less monomorphs is always better, and we could save some memory with this, I guess.

_(ideally we of course want a bottom type but that's a different topic)_

**Comment by nadako** [comment]
> I'm wondering if we should be a bit more clever and type all terminators as Void if they are used in WithType.NoValue context.

Tried that quickly and ran into problems with inlined returns, so it needs more investigation.

**Comment by Simn** [comment]
@ncannasse You wouldn't happen to remember why this was typed as `Dynamic`? I recall discussing this with you like 10 years ago and it made sense to me at the time, but now I'm not really seeing why this should be the case.

**Comment by Simn** [comment]
Looking at this again, I wonder if we should only create monomorphs for this when we're typing a value. I don't see what we gain by creating monos for any block-level `return` expression and such. I'm actually trying to have _less_ wild monos floating around, so this would work a bit against that.

What do you think?

**Comment by nadako** [comment]
Makes sense. I think I've tried that but had some issue with inlining and didn't have time to figure it out since then...

File: src/typing/typer.ml [hunk 1]
```diff
@@ -36,6 +36,12 @@ open Operators
 (* ---------------------------------------------------------------------- *)
 (* TOOLS *)
 
+let mono_or_dynamic ctx with_type p = match with_type with
+	| WithType.NoValue ->
+		t_dynamic
```
Comment (nadako) [inline src/typing/typer.ml:41]: should it be Void maybe?

Comment (Simn) [reply src/typing/typer.ml:41]: I'm a bit worried about some crazy inline situation where we end up with Void in a value-place... With Dynamic we at least have the same behavior as before.

**Pull request merged**